collect_app/src/main/java/org/odk/collect/android/tasks/DownloadFormsTask.java

```diff
@@ -93,7 +94,7 @@ private boolean isXformsManifestNamespacedElement(Element e) {
         Collect.getInstance().getActivityLogger().logAction(this, "downloadForms",
                 String.valueOf(total));
 
-        HashMap<FormDetails, String> result = new HashMap<FormDetails, String>();
+        final HashMap<FormDetails, String> result = new HashMap<>();
```

Comment:
Technically this could be a `Map` as well, but not a serious issue.

Reply:
Somewhere I found that a concrete HashMap implementation was required, so couldn’t change to use the Map interface.

Reply:
Yeah, the concrete implementations are littered all over the app so it can be a pain to change them.